### PR TITLE
Add dbSendManip() generic

### LINKED ISSUE #20: Add dbSendUpdate or equivalent to DBI
Please add a function like RJDBC::dbSendUpdate to DBI so that it will be available for other db access packages such as ROracle. See the R-sig-DB thread starting with https://stat.ethz.ch/pipermail/r-sig-db/2014q3/001443.html for more information and discussion, including alternate name and some implementation details.

### PR DESCRIPTION
**Pull request by krlmlr**
- Expects a `DBIresult` as return value.
- Fix and link documentation of `dbSendQuery()`, `dbGetQuery()` and `dbExecute()`.
- Now called by `dbExecute()`.
- Implement default `dbSendUpdate()` that forwards to `dbSendQuery()`.
- Includes updated revdep check results.

@hannesmuehleisen @imanuelcostigan @s-u: Your packages already define `dbSendUpdate()`, which is fine, but the new interface expects a `DBIresult` which can be queried for the number of rows affected, and closed. It will be also used for prepared statements.

I understand that this is a breach of the contract which has been defined and established earlier by your packages. MonetDBLite checks fail because of that, too. Perhaps it would be safest if DBI used a different method name, I'd love to hear your opinion here.

Fixes #20.

CC @bborgesr.

### CODE REVIEW

**Comment by hannesmuehleisen** [comment]
happy to change my implementation to match the new behaviour.

**Comment by krlmlr** [comment]
Just saw that `dbSendUpdate()` in RJDBC is more than 8 years old. At this point I'd argue that it's not safe anymore to change the contract. I'll rename the function.

What should the new name be, `dbSendManip()`?

**Comment by imanuelcostigan** [comment]
@krlmlr I'll take a look over next week and revert with comments if that's ok?

**Comment by krlmlr** [comment]
@imanuelcostigan: Thanks. I think I'm not going to define `dbSendUpdate()` but a differently named function, current candidate `dbSendManip()`. The difference is the return value, `dbSendManip()` needs a `DBIresult`. I'm happy to take naming suggestions.

**Comment by imanuelcostigan** [comment]
Concern I have is that you are expecting `dbSendManip()` to return a `DBIResult` (not `DBIresult` I assume). I don't think non-query statements return this for my backend - but I need to take a closer look. 

ON naming - not convinced with naming of `dbSendManip()`! I'll see if I can figure out anything better.

**Comment by krlmlr** [comment]
Yes, returning a `DBIResult` would be a change you'd need to make to your backend to be "fully DBI compatible". RJDBC currently returns NULL in `dbSendUpdate()`, that's why I'm renaming the function in the first place. Perhaps this even should be changed in RJDBC itself.

**Comment by bborgesr** [comment]
@krlmlr, I'm also a little confused as to why there is a need for this (https://github.com/rstats-db/DBI/issues/20#issuecomment-230799066) -- `dbExecute` statements tend to be more atomic than query statements by nature, so I think there's less of a case for the symmetry argument... Is it so bad that `dbExecute` calls `dbSendQuery` instead of having a dedicated function for execute statements? It feels a little unnecessary to me, but obviously I was not part of this discussion from the beginning, so I may be completely off base here... I do worry about this being too many too closely related things for users to deal with in one release -- the potential for confusion seems high... Again, apologies if I'm just being nit-picky here...

**Comment by krlmlr** [comment]
@bborgesr: The motivation for `dbSendManip()` is https://github.com/rstats-db/DBI/issues/20#issuecomment-230805023, especially the first bullet point. Let me rephrase:

For prepared queries, you'd:

1. call `dbSendQuery()` with a query that contains placeholders
2. call `dbBind()` to assign values to the placeholders
3. call `dbFetch()` to retrieve query results
4. repeat 2 and 3 as needed
5. call `dbClearResult()`.

I'd like to use the same pattern for prepared data manipulation statements such as `INSERT INTO t VALUES (?, ?)`.

I agree that a monolith `dbExecute()` may be simpler to implement for some backends. However, for RSQLite, RMySQL and RPostgres, `dbSendQuery()` and `dbSendManip()` are currently the same functions -- they work out of the box. All there's to do for MonetDB and RJDBC/RSQLServer (who originally had `dbSendUpdate()`) is to wrap the number of rows affected in a `DBIResult` object. I think this isn't too much asked for.

Still open for naming suggestions.

**Comment by bborgesr** [comment]
Ok, I get it a bit better now. You probably thought about this already, but `dbSendStatement`? or `dbSendExec` (which would stand for "send executable statement" -- mirroring `dbExecute`)? This would have the advantage that it is a bit more general than `dbSendManip`, since data manipulation statements aren't the only thing that's supported here, right? You could also use it to set connection variables and stuff like that, which is not actually manipulating any data...

**Comment by krlmlr** [comment]
I like `dbSendStatement()`. Yes, it's for all kinds of non-SELECT statements.
